EosPayload/lib/orcheostrator/orcheostrator.py

```diff
@@ -77,71 +77,50 @@ def terminate(self) -> None:
                 driver.process.terminate()
                 driver.update_status(Status.TERMINATED)
 
-    @staticmethod
-    def valid_driver(driver) -> bool:
-        """ Determines if given class is a valid driver.
-
-        :param driver: the class in question
-        :return: True if valid, otherwise False
-        """
-        return (
-            (driver is not None)
-            and inspect.isclass(driver)
-            and issubclass(driver, DriverBase)
-            and driver.__name__ not in ["DriverBase", "PositionAwareDriverBase"]
-        )
-
     #
     # PROTECTED HELPER METHODS
     #
 
     def _spawn_drivers(self) -> None:
         """ Enumerates over all the classes in the drivers dir and spins them up """
         self._logger.info("Spawning Drivers")
-        for attribute_name in dir(drivers):
-            driver = getattr(drivers, attribute_name)
-            if self.valid_driver(driver):
-                container = DriverContainer(driver)
-                try:
-                    if driver.get_device_id() is None:
-                        self._logger.error(f"can't spawn process for device from class '{driver.__name__}'"
-                                           " because device id is not defined")
-                        container.update_status(Status.INVALID)
-                        self._drivers['<' + driver.__name__ + '>'] = container
-                        continue
-                    if not driver.enabled():
-                        self._logger.warning(f"skipping device '{driver.get_device_pretty_id()}' from"
-                                             f" class '{driver.__name__}' because it is not enabled")
-                        container.update_status(Status.DISABLED)
-                        self._drivers[driver.get_device_id()] = container
-                        continue
-                    self._logger.info(f"spawning process for device '{driver.get_device_pretty_id()}' from"
-                                      f" class '{driver.__name__}'")
-                    proc = Process(target=self._driver_runner, args=(driver, self.output_directory), daemon=True)
-                    container.process = proc
-                    container.update_status(Status.HEALTHY, 1)
-                    proc.start()
-                    self._drivers[driver.get_device_id()] = container
-                except Exception as e:
-                    self._logger.critical("A fatal exception occurred when attempting to load driver from"
-                                          f" class '{driver.__name__}': {e}\n{traceback.format_exc()}")
+        driver_list = self.orcheostrator_config.enabled_devices
+
+        for container in driver_list:
+            driver = container.driver
+            driver_config = container.config
+            try:
+                if driver_config.get("device_id") is None:
```

Comment:
it this even possible at this point?  This should cause an error in config parsing

Reply:
Yeeted